### ISSUE TITLE
Incorrect documentation for `fit_transform` in `sklearn_api` API 

### ISSUE THREAD

**Issue by danielchalef** [RaRe-Technologies/gensim#2753]
#### Problem description

`fit_transform`  in the `sklearn_api` is documented to return:
`X_new : numpy array of shape [n_samples, n_features_new]`

This is not the case. What is returned is:
` iterable of list (int, float) 2-tuples.`

It appears `fit_transform` is inherited from `sklearn.base.TransformerMixin` and therefore needs to be documented appropriately (or the method overridden).

#### Steps/code/corpus to reproduce

```python
model2 = TfIdfTransformer(dictionary=dct)
vecs = model2.fit_transform(corpus)
type(vecs)
> list
```

#### Versions
```
Linux-4.14.146-119.123.amzn2.x86_64-x86_64-with-debian-buster-sid
Python 3.7.6 (default, Jan  8 2020, 19:59:22) 
[GCC 7.3.0]
NumPy 1.18.1
SciPy 1.3.2
gensim 3.8.0
FAST_VERSION 1
```

**Comment by amirothman** [comment]
this seems like a sklearn issue rather than gensim.

**Comment by piskvorky** [comment]
Yeah, there has been some discussion around the purpose (and therefore API) of these sklearn wrappers in Gensim. See also #2825.

**Comment by danielchalef** [comment]
I was looking for a fast, memory-efficient multiprocessing tfidf transformer. I’ve since moved on from that project and this is no longer needed. The issue was more a heads up regarding documentation than a feature request. Thanks!

**Comment by piskvorky** [comment]
Thanks.

IMO it's clear that getting rid of the Gensim streaming capability (converting from/to numpy/scipy matrices) would throw the baby out with the bath water. It would be kosher with the sklearn API, but absolutely useless to users.

IMO we should either figure out a way to support streaming (at least within the portion of the sklearn pipeline that deals with gensim), or drop the wrappers altogether.
